routes/auth.routes.js

```diff
@@ -0,0 +1,71 @@
+const { Router } = require("express");
+const router = new Router();
+const mongoose = require("mongoose");
+const User = require("../models/User.model");
+const bcryptjs = require("bcryptjs");
+const saltRounds = 10;
+const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard");
+
+router.get("/signup", isLoggedOut, (req, res) => res.render("auth/signup"));
+
+router.post("/signup", (req, res, next) => {
+  const { username, password } = req.body;
+
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({
+        username,
+        password: hashedPassword,
+      });
+    })
+    .then((userFromDB) => {
+      console.log("Newly created user is: ", userFromDB);
```

Comment:
After a user is created, you can redirect to somewhere else, for example to the login page